pkg/handler/enqueue.go

```diff
@@ -52,25 +53,47 @@ func (e *TypedEnqueueRequestForObject[T]) Create(ctx context.Context, evt event.
 		enqueueLog.Error(nil, "CreateEvent received with no metadata", "event", evt)
 		return
 	}
-	q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+
+	item := reconcile.Request{NamespacedName: types.NamespacedName{
 		Name:      evt.Object.GetName(),
 		Namespace: evt.Object.GetNamespace(),
-	}})
+	}}
+
+	priorityQueue, isPriorityQueue := q.(priorityqueue.PriorityQueue[reconcile.Request])
+	if !isPriorityQueue {
+		q.Add(item)
+		return
+	}
+	addToPriorityQueueCreate(priorityQueue, evt, item)
 }
 
 // Update implements EventHandler.
 func (e *TypedEnqueueRequestForObject[T]) Update(ctx context.Context, evt event.TypedUpdateEvent[T], q workqueue.TypedRateLimitingInterface[reconcile.Request]) {
 	switch {
 	case !isNil(evt.ObjectNew):
-		q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+		item := reconcile.Request{NamespacedName: types.NamespacedName{
```

Comment:
Could we also do the same for `Funcs`? That is the only remaining handler

Reply:
Not sure if this is how you wanted this done but any feedback is appreciated.

Reply:
I think in `Funcs` we have to do the same dance as in `TypedEnqueueRequestForObject` because we export the type and not a constructor, otherwise the original goal of `get this by default` won't be achieved

Reply:
Makes sense, I did what we did with `TypedEnqueueRequestForObject` because I can't add methods to the `Funcs` type.  So I did the same thing we did with this.